sparse_ho/models.py

```diff
@@ -278,20 +292,26 @@ class WeightedLasso():
         Data.
     y: {ndarray, sparse matrix} of (n_samples)
         Target
-    estimator: instance of ``sklearn.base.BaseEstimator``
-        An estimator that follows the scikit-learn API.
     log_alpha_max: float
         logarithm of alpha_max if already precomputed
     """
 
     def __init__(
-            self, X, y, max_iter=1000, estimator=None, log_alpha_max=None):
-        self.X = X
-        self.y = y
+            self, criterion, algo, max_iter=1000, log_alpha_max=None):
+        self.criterion = criterion
+        self.algo = algo
         self.max_iter = max_iter
-        self.estimator = estimator
+        self._estimator = Lasso(warm_start=True)
```

Comment:
this should happen in the fit to be sklearn compliant. The init just set attributes untouched.

So you would do

self.estimator = estimator

here and in the fit

`self.estimator_ = clone(self.estimator) if self.estimator is not None else Lasso(warm_start=True)`

Reply:
I would like to avoid passing the estimator in the init
Could I do `self.estimator = None` here ?